assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py

```diff
@@ -0,0 +1,263 @@
+
+
+def read_json_content(path: str) -> str:
+    """Read json content from path."""
+    if not path:
+        return None
+    if _is_local_path(path):
+        with open(path) as f:
+            json_str = f.read()
+        return json_str
+    hdfs_path, container_client = get_hdfs_path_and_container_client(path)
+    idx = hdfs_path.find('.core.windows.net/') + len('.core.windows.net/')
+    file_path = hdfs_path[idx:]
+    if not container_client:
+        raise ValueError(f"Fail to get credential to access dataref file {hdfs_path}")
+    elif isinstance(container_client, FileSystemClient):
+        with container_client.get_file_client(file_path) as file_client:
+            json_bytes = file_client.download_file().readall()
+    else:  # must be ContainerClient
+        with container_client.get_blob_client(file_path) as blob_client:
+            json_bytes = blob_client.download_blob().readall()
+
+    return json_bytes.decode()
+
+
+def _get_local_file_list(start_datetime: datetime, end_datetime: datetime, input_data: str) -> List[str]:
+    file_list = []
+    root_folder = input_data.rstrip('/')
+    if end_datetime.minute == 0 and end_datetime.second == 0:
+        # if end_datetime is a whole hour, the last hour folder is not needed
+        end_datetime -= timedelta(seconds=1)
+    cur_datetime = start_datetime
+    while cur_datetime <= end_datetime:
+        cur_folder = f"{root_folder}/{cur_datetime.strftime('%Y/%m/%d/%H')}"
+        if os.path.isdir(cur_folder):
+            for file in os.listdir(cur_folder):
+                full_qualify_file = os.path.join(cur_folder, file)
+                if os.path.isfile(full_qualify_file) and file.endswith(".jsonl"):
+                    file_list.append(full_qualify_file)  # local winutils/hadoop package doesn't support wildcard
+        cur_datetime += timedelta(hours=1)
+    return file_list
+
+
+def _is_local_path(path: str) -> bool:
+    if not path:
+        return False
+    return os.path.isdir(path) or os.path.isfile(path) or path.startswith("file://") or path.startswith("/") \
+        or path.startswith(".") or re.match(r"^[a-zA-Z]:[/\\]", path)
+
+
+def _folder_exists(container_client: Union[ContainerClient, FileSystemClient], folder_path: str) -> bool:
+    """Check if hdfs folder exists."""
+    if isinstance(container_client, FileSystemClient):
+        return container_client.get_directory_client(folder_path).exists()
+    else:
+        blobs = container_client.list_blobs(name_starts_with=folder_path)
+        return any(blobs)
+
+
+def _get_datastore_and_path_from_azureml_path(azureml_path: str) -> Tuple[str, str]:
+    start_idx = azureml_path.find('/datastores/')
+    end_idx = azureml_path.find('/paths/')
+    return azureml_path[start_idx+12:end_idx], azureml_path[end_idx+7:]
+
+
+def _get_container_client(datastore: Union[AzureDataLakeGen2Datastore, AzureBlobDatastore]) \
```

Comment:
I get a little confused, they are both file system client, but one is generated from http/abfs path, the other is generated from azureml datastore path, right?